com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/textureset/TextureSetGenerator.java

```diff
@@ -263,9 +263,21 @@ private static SpriteGeometry.Builder createSpriteGeometryFromRect(Rect rect) {
 
         int imageWidth = rect.getWidth();
         int imageHeight = rect.getHeight();
-        builder.setWidth(imageWidth);
-        builder.setHeight(imageHeight);
-        builder.setRotated(rect.getRotated());
+        boolean rotated = rect.getRotated();
+        builder.setRotated(rotated);
+
+        if (rotated)
+        {
+            // for legacy reasons, we need to rotate it back
+            // The geometry wants the size in unrotated form
+            builder.setWidth(imageHeight);
+            builder.setHeight(imageWidth);
+        }
+        else
+        {
+            builder.setWidth(imageWidth);
+            builder.setHeight(imageHeight);
+        }
```

Comment:
eventually, I'd like to centralize these things even more, but for now I think it's good enough.